### ISSUE TITLE
The `markdown-gren` package breaks with `0.4.5`

### ISSUE THREAD

**Issue by dwyd** [gren-lang/compiler#270]
Hello. I original posted this in https://github.com/icidasset/markdown-gren/issues/2 but seems it belongs here. This issue happens with Gren version `0.4.5` but the package seems to work fine with version `0.4.4`.

The following error happened when running the [icidasset/shikensu-gren](https://github.com/icidasset/shikensu-gren/blob/main/example/src/Main.gren) example code using Gren version `0.4.5` (which I traced to the `icidasset/markdown-gren` package)...

```
-- ERROR -----------------------------------------------------------------------

I ran into something that bypassed the normal error reporting process! I
extracted whatever information I could from the internal error:

>   ["parse","parser","rawToBlocks","translateRaw","translateRawElement"]
>   CallStack (from HasCallStack):
>     error, called at compiler/src/Generate/JavaScript.hs:64:15 in main:Generate.JavaScript

These errors are usually pretty confusing, so start by asking around on one of
forums listed at https://gren-lang.org/community to see if anyone can get you
unstuck quickly.

-- REQUEST ---------------------------------------------------------------------

If you are feeling up to it, please try to get your code down to the smallest
version that still triggers this message. Ideally in a single Main.gren and
gren.json file.

From there open a NEW issue at https://github.com/gren-lang/compiler/issues with
your reduced example pasted in directly. (Not a link to a repo or gist!) Do not
worry about if someone else saw something similar. More examples is better!

This kind of error is usually tied up in larger architectural choices that are
hard to change, so even when we have a couple good examples, it can take some
time to resolve in a solid way.gren: ["parse","parser","rawToBlocks","translateRaw","translateRawElement"]
CallStack (from HasCallStack):
  error, called at compiler/src/Generate/JavaScript.hs:64:15 in main:Generate.JavaScript
```

I also tried cloning the `icidasset/markdown-gren` repo and running the tests with Gren `0.4.5` but that produces this error...

```
-- ERROR -----------------------------------------------------------------------

I ran into something that bypassed the normal error reporting process! I
extracted whatever information I could from the internal error:

>   ["emphasisWithUnderscoreParser","emphasisWithStarParser","emphasisParser","parse","parser","rawParser","rawParserLoop"]
>   CallStack (from HasCallStack):
>     error, called at compiler/src/Generate/JavaScript.hs:64:15 in main:Generate.JavaScript

These errors are usually pretty confusing, so start by asking around on one of
forums listed at https://gren-lang.org/community to see if anyone can get you
unstuck quickly.

-- REQUEST ---------------------------------------------------------------------

If you are feeling up to it, please try to get your code down to the smallest
version that still triggers this message. Ideally in a single Main.gren and
gren.json file.

From there open a NEW issue at https://github.com/gren-lang/compiler/issues with
your reduced example pasted in directly. (Not a link to a repo or gist!) Do not
worry about if someone else saw something similar. More examples is better!

This kind of error is usually tied up in larger architectural choices that are
hard to change, so even when we have a couple good examples, it can take some
time to resolve in a solid way.gren: ["emphasisWithUnderscoreParser","emphasisWithStarParser","emphasisParser","parse","parser","rawParser","rawParserLoop"]
CallStack (from HasCallStack):
  error, called at compiler/src/Generate/JavaScript.hs:64:15 in main:Generate.JavaScript
```

The issue seems to be happening in the [rawParserLoop](https://github.com/icidasset/markdown-gren/blob/0656ef10e245af3944c31c404936b744d33c5a44/src/Markdown/Parser/Inlines.gren#L145-L228) call.

Lastly, thanks for building Gren!

**Comment by robinheghan** [comment]
Thank you for this detailed bug report! I'll take a look on monday with the hopes of fixing it in time for the release of Gren 24W.

**Comment by dwyd** [comment]
No problem and sounds good! Wish I could help more but it's unfamiliar territory for me 😊

**Comment by robinheghan** [comment]
Fixed on the `main` branch. Will be released with v0.5.
